src/components/HorizonalCheckboxField.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC } from 'react';
+import { css, cx } from 'emotion';
+import { useStyles, Checkbox, Label } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { useFormContext } from 'react-hook-form';
+
+const getStyles = (theme: GrafanaTheme) => ({
+  container: css`
+    display: flex;
+    align-items: center;
+    margin-bottom: ${theme?.spacing.md};
+  `,
+  checkbox: css`
+    position: relative;
+  `,
+  label: css`
+    margin-bottom: 0;
+  `,
+});
+
+interface Props {
+  disabled?: boolean;
+  id?: string;
+  name?: string;
+  label?: string;
+  description?: string;
+  className?: string;
+  value?: boolean;
+  onChange?: () => void;
```

Comment:
Are we sure all these props should be optional?
I mean, in the way how we are going to use this component we most probably always want to specify an `id` and a `name` at least, right?

Reply:
Good question! The `name` prop sort of implies registering with a form, but that won't always be the case. The `id` prop should probably be required, though.